app/forms/journeys/additional_payments_for_teaching/qualification_form.rb

```diff
@@ -0,0 +1,62 @@
+module Journeys
+  module AdditionalPaymentsForTeaching
+    class QualificationForm < Form
+      QUALIFICATION_OPTIONS = %w[
+        postgraduate_itt
+        undergraduate_itt
+        assessment_only
+        overseas_recognition
+      ].freeze
+
+      attribute :qualification, :string
+
+      validates :qualification,
+        inclusion: {
+          in: QUALIFICATION_OPTIONS,
+          message: "Select the route you took into teaching"
+        }
+
+      def initialize(journey:, claim:, params:)
+        super
+
+        self.qualification = permitted_params.fetch(
+          :qualification,
+          claim.eligibility.qualification
+        )
+      end
+
+      def save
+        return false unless valid?
+
+        # We set the attribute like this, rather than using `update!` from the
+        # superclass, as we need "qualification" to be in the `Eligibility#changed`
+        # list of attributes for the `reset_dependent_answers` method to work
+        claim.assign_attributes(
+          eligibility_attributes: {qualification: qualification}
+        )
+        claim.reset_eligibility_dependent_answers(["qualification"])
+        claim.save!
+
+        true
+      end
+
+      def save!
+        raise ActiveRecord::RecordInvalid.new unless save
+      end
+
+      def backlink_path
+        Rails
+          .application
+          .routes
+          .url_helpers
+          .claim_path(params[:journey], page_sequence.previous_slug)
+      end
```

Comment:
Yeah I think that makes sense 👍

Reply:
Thanks, I think it's what we are doing in the [controller](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/fb4c8e1ec3fd21b19aa4433aaa3f568d8b766ce3/app/controllers/claims_controller.rb#L170-L173), so it's worth having the same default behaviour unless we need really to override it in the subclass (not the norm). This would also fix it for all the new forms (that are failing to show the backlink if submitted but an error occurs)

What do you think about a small test in the base class? Smth like:

```ruby
...
  let(:params) { ActionController::Parameters.new({journey: "test-journey", slug: "test_slug", claim: claim_params}) }
...
  describe "#backlink_path" do
    before do
      allow_any_instance_of(Journeys::PageSequence).to receive(:previous_slug)
        .and_return(previous_slug)
    end

    describe "when the previous slug is present" do
      let(:previous_slug) { "previous-slug" }

      it { expect(form.backlink_path).to eq("/test-journey/previous-slug") }
    end

    describe "when the previous slug is not present" do
      let(:previous_slug) { nil }

      it { expect(form.backlink_path).to be_nil }
    end
  end
```